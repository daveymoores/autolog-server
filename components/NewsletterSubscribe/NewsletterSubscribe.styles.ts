import css from "styled-jsx/css";

export default css`
  .newsletter-subscribe {
    display: flex;
    align-items: stretch;
    justify-content: center;
    flex-direction: column;
    height: 100%;
  }

  .newsletter-subscribe form {
    display: flex;
    align-items: center;
    justify-content: center;
    flex-direction: column;
  }

  .newsletter-subscribe h3 {
    text-align: center;
    font-size: 18px;
    font-weight: 500;
    margin-bottom: 25px;
  }
`;
