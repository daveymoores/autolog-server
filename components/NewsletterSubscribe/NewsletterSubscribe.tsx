import React from "react";

import Button from "../Button/Button";
import Input from "../Input/Input";
import styles from "./NewsletterSubscribe.styles";

const NewsletterSubscribe: React.FC = () => (
  <>
    <div className="newsletter-subscribe">
      <h3>Subscribe to be the first to know about updates</h3>
      <form>
        <Input />
        <Button type="submit" text="subscribe" />
      </form>
    </div>
    <style jsx>{styles}</style>
  </>
);

export default NewsletterSubscribe;
