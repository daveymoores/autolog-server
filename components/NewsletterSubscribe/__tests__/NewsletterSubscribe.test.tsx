import React from "react";
import renderer from "react-test-renderer";

import NewsletterSubscribe from "../NewsletterSubscribe";

describe("NewsletterSubscribe", () => {
  it("renders NewsletterSubscribe unchanged", () => {
    const tree = renderer.create(<NewsletterSubscribe />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
