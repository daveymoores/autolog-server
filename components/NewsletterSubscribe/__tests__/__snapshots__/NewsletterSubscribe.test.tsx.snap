// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsletterSubscribe renders NewsletterSubscribe unchanged 1`] = `
<div
  className="jsx-3185220219 newsletter-subscribe"
>
  <h3
    className="jsx-3185220219"
  >
    Subscribe to be the first to know about updates
  </h3>
  <form
    className="jsx-3185220219"
  >
    <input
      className="jsx-3861775726 t-gen--cell"
      type="text"
    />
    <button
      className="py-2 px-4 bg-gradient-to-tl from-indigo-500 to-indigo-700 text-white text-md rounded-md shadow focus:outline-none"
      type="submit"
    >
      subscribe
    </button>
  </form>
</div>
`;
